### PR TITLE
QoL addition and adjustment to squad role vendors.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
Change how section order to make it more standardize and hopefully more new user friendly.
add a new section binoculars and generalize the usage to engineering supply, primary ammunition and sidearm ammunition 

Main balance change is the cost of sandbag from 20 to 15 and making it accessible to every squad role also added a stack of 5 barbed wire for 5 point

# Explain why it's good for the game

This PR is mainly to improve QoL and go to standardize how squad role propose to each role as much as possible.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
qol: Change order of section in squad roles vendors to put clothing and armor first then medical and engineering.
qol: add the IO to the new squad role vendor standard.
qol: separate ammunition section into primary and sidearm ammunition.
add: added an engineering section to all squad vendor role with sandbag barbed wire, E-tool, Fuel Canister and ME3 hand welder.
code: made binoculars be their own section for all squad vendor role.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
i will update changelog and all when if fixed it. if you know what's wrong tell me...

**Review by Birdtalon — COMMENTED** [review]
Commas.

**Review by Drulikar — CHANGES_REQUESTED** [review]
Please stick to maintaining the same access to items in a pr like this, especially with how ambiguous the changelog is. Remove all changes (additional access to NVG) that result in this requiring balance considerations. Keep this as a quality of life update. If you want additional access to NVG to be considered, have that be a PR all by itself.

File: code/game/machinery/vending/vendor_types/intelligence_officer.dm [hunk 1]
```diff
@@ -36,6 +35,54 @@ GLOBAL_LIST_INIT(cm_vending_gear_intelligence_officer, list(
 		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
 		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
 		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/intelligence_officer.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm [hunk 2]
```diff
@@ -20,12 +33,31 @@ GLOBAL_LIST_INIT(cm_vending_gear_engi, list(
...
+		list("M39 AP Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/ap , null, VENDOR_ITEM_REGULAR),
+		list("M39 Extended Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/extended , null, VENDOR_ITEM_REGULAR),
+		list("M41A AP Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/ap , null, VENDOR_ITEM_REGULAR),
+		list("M41A Extended Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/extended , null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm [hunk 3]
```diff
@@ -74,24 +52,58 @@ GLOBAL_LIST_INIT(cm_vending_gear_leader, list(
...
+		list("M41A AP Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/ap , null, VENDOR_ITEM_REGULAR),
+		list("M41A Extended Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/extended , null, VENDOR_ITEM_REGULAR),
+		list("M240 Incinerator Tank (Napthal)", 3, /obj/item/ammo_magazine/flamer_tank, null, VENDOR_ITEM_REGULAR),
+		list("M240 Incinerator Tank (B-Gel)", 3, /obj/item/ammo_magazine/flamer_tank/gellied, null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm [hunk 4]
```diff
@@ -47,6 +58,30 @@ GLOBAL_LIST_INIT(cm_vending_gear_medic, list(
...
+		list("M39 AP Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/ap , null, VENDOR_ITEM_REGULAR),
+		list("M39 Extended Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/extended , null, VENDOR_ITEM_REGULAR),
+		list("M41A AP Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/ap , null, VENDOR_ITEM_REGULAR),
+		list("M41A Extended Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/extended , null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm [hunk 5]
```diff
@@ -45,11 +45,49 @@ GLOBAL_LIST_INIT(cm_vending_clothing_marine, list(
...
+		list("M39 AP Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/ap , null, VENDOR_ITEM_REGULAR),
+		list("M39 Extended Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/extended , null, VENDOR_ITEM_REGULAR),
+		list("M41A AP Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/ap , null, VENDOR_ITEM_REGULAR),
+		list("M41A Extended Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/extended , null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm [hunk 6]
```diff
@@ -4,14 +4,36 @@ GLOBAL_LIST_INIT(cm_vending_gear_smartgun, list(
...
+		list("SU-6 Smart Pistol", 12, /obj/item/storage/box/guncase/smartpistol, null, VENDOR_ITEM_REGULAR),
+
 		list("SMARTGUN AMMUNITION", 0, null, null, null),
 		list("M56 Smartgun Drum", 15, /obj/item/ammo_magazine/smartgun, null, VENDOR_ITEM_RECOMMENDED),
 
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm [hunk 7]
```diff
@@ -106,15 +106,38 @@ GLOBAL_LIST_INIT(cm_vending_clothing_specialist, list(
...
 
+		list("RESTRICTED FIREARMS", 0, null, null, null),
+		list("VP78 Pistol", 8, /obj/item/storage/box/guncase/vp78, null, VENDOR_ITEM_REGULAR),
+		list("SU-6 Smart Pistol", 12, /obj/item/storage/box/guncase/smartpistol, null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
+		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
+		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm]: done thanks

File: code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm [hunk 8]
```diff
@@ -1,16 +1,44 @@
...
 		list("M39 AP Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/ap , null, VENDOR_ITEM_REGULAR),
 		list("M39 Extended Magazine (10x20mm)", 10, /obj/item/ammo_magazine/smg/m39/extended , null, VENDOR_ITEM_REGULAR),
 		list("M41A AP Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/ap , null, VENDOR_ITEM_REGULAR),
 		list("M41A Extended Magazine (10x24mm)", 10, /obj/item/ammo_magazine/rifle/extended , null, VENDOR_ITEM_REGULAR),
+
+		list("SIDEARM AMMUNITION", 0, null, null, null),
 		list("M44 Heavy Speed Loader (.44)", 10, /obj/item/ammo_magazine/revolver/heavy, null, VENDOR_ITEM_REGULAR),
 		list("M4A3 HP Magazine", 5, /obj/item/ammo_magazine/pistol/hp, null, VENDOR_ITEM_REGULAR),
 		list("M4A3 AP Magazine", 5, /obj/item/ammo_magazine/pistol/ap, null, VENDOR_ITEM_REGULAR),
+		list("VP78 Magazine", 5, /obj/item/ammo_magazine/pistol/vp78, null, VENDOR_ITEM_REGULAR)
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm]: done thanks

**Review by realforest2001 — CHANGES_REQUESTED** [review]
As with the comment above regarding NVGs, please split all the balance changes out of this PR for atomisation purposes.

**Comment by Huffie56** [comment]
going to do a PR way smaller and atomized...

**Pull request closed without merging**